### ISSUE TITLE
[Tagging] Discovery: Move the library-authoring MFE into the course-authoring MFE

### ISSUE THREAD

**Issue by bradenmacdonald** [openedx/modular-learning#164]
We want the "Tag Drawer" component to work identically in the course authoring experience and the library authoring experience.

There are two ways to achieve this:
1. Move the "Tag Drawer" to its own repo, so that the [Course Authoring MFE](https://github.com/openedx/frontend-app-course-authoring/) and [Library Authoring MFE](https://github.com/openedx/frontend-app-library-authoring) can both include it as a component. Or,
2. Consolidate the [Library Authoring MFE](https://github.com/openedx/frontend-app-library-authoring) into the [Course Authoring MFE](https://github.com/openedx/frontend-app-course-authoring/)

The benefits of either approach would be:
* The Tag Drawer no longer needs to load in an iframe, so loads much faster
* The Unit/Library page can use react-query to display the number of tags, so the number of tags will be instantly and automatically updated whenever changes are made to a component's tags.

For this ticket:
1. Try a quick prototype of "Option 2" (consolidate) to get a sense of how much work it is.
2. Get input from @arbrandes and @KristinAoki about which approach is preferable.
3. Create tickets for implementing that approach.

**Comment by KristinAoki** [comment]
TNL would prefer option 2. It is an idea that the team been tossing around in greater frequency as both repos continue to grow and have many overlapping components.

**Comment by rpenido** [comment]
Hi!

@KristinAoki, could you please help us with your opinion here? :smiley: 

I made a prototype of the option 2 here:
- https://github.com/openedx/frontend-app-course-authoring/pull/774

From a technical standpoint, I didn't find any major issues. We will benefit significantly from reusing components (and we can't do it gradually, removing the duplicated components).

I'm not sure how we should coordinate it with the community. Have we done something similar before?

The worst scenario would be to do the merge and end up with two concurring library-authoring features! :sweat_smile:

**Comment by connorhaugh** [comment]
@jesperhodge has done some preliminary investigation and created some tasks for this effort, He should share those with the community

**Comment by rpenido** [comment]
Hi @jesperhodge! Thank you for your response.
I think my prototype is in the same state as yours. I did it to gather the complexity and plan some steps. I'm not sure if I will be able to continue from here.

My task list is pretty much the same as yours. I am happy that we converged here (a sign that we are on the right path)! I just included some tasks to handle the OEP-21 deprecation process, and I think the course-import reducer task (and some pipeline adjustments) could be done earlier in the library-authoring repo before the actual merge (to reduce the merge complexity a bit). I also think that consolidating the header (and other duplicated components) could be handled in a cleanup task after the merge (also, seeking to reduce complexity).

We are also interested in the merge, but, as I said, we don't have a timeline or the resources to handle it right now. I will keep following this and hope we can help get this done in the medium term.
